bin/i18n/redact_restore_utils.rb

```diff
@@ -63,40 +63,33 @@ def self.restore_data(source_data, redacted_data, plugins = [], format = 'md')
     return JSON.parse(restored)
   end
 
+  # Restores redacted data into a json/yaml/yml file
+  #
+  # @param source [String] the source file path
+  # @param redacted [String] the redacted file path
+  # @param dest [String] the target (restored) file path
+  # @param plugins [Array] the list of plugins to use
+  # @param format [String] the format of the data (md, txt, etc.)
   def self.restore(source, redacted, dest, plugins = [], format = 'md')
-    return unless File.exist?(source)
-    return unless File.exist?(redacted)
-
-    if I18nScriptUtils.json_file?(source)
-      source_data = I18nScriptUtils.parse_file(source)
-      return unless source_data
-
-      redacted_data = I18nScriptUtils.parse_file(redacted)
-      return unless redacted_data
+    raise "[#{source}] source file does not exist" unless File.exist?(source)
+    raise "[#{redacted}] redacted file does not exist" unless File.exist?(redacted)
 
-      restored = RedactRestoreUtils.restore_data(source_data, redacted_data, plugins, format)
-      I18nScriptUtils.write_json_file(dest, restored)
-      return
-    end
+    source_data = I18nScriptUtils.parse_file(source) || {}
+    raise "[#{source}] source data does not exist" if source_data.empty?
 
-    # yaml files have a locale key wrapping all content. Redacted files have the translated language key, while the
-    # un-redacted file keeps the English key. We need to extract the data inside the locale key to restore content.
-    if I18nScriptUtils.yaml_file?(source)
-      source_data = I18nScriptUtils.parse_file(source)
-      return unless source_data
-      return unless source_data&.values&.first&.length
+    redacted_data = I18nScriptUtils.parse_file(redacted) || {}
+    return if redacted_data.empty?
 
-      redacted_data = I18nScriptUtils.parse_file(redacted)
-      return unless redacted_data
-      return unless redacted_data&.values&.first&.length
+    # Some yaml files have a locale key wrapping all content.
+    # Redacted files have the translated language key, while the un-redacted file keeps the English key.
+    # We need to extract the data inside the locale key to restore content.
+    source_data[redacted_data.keys.first] = source_data.delete(source_data.keys.first) if source_data.keys.size == 1
```

Comment:
What would happen if we have a json file with just one string?

Reply:
1. If the file data is only one string: it will raise the error "undefined method `keys` for String".
That's okay, we shouldn't have such situations, but if we do, it will mean that something is wrong anyway.
2. If the file data is a hash with one root key:  it will replace the root key of the source data with one from the restored data. That's also fine, since it should be the same

Reply:
1. If the file data is only one string: it will raise the error "undefined method keys for String".
- That's okay, we shouldn't have such situations, but if we do, it will mean that something is wrong anyway.
restore/redact js libs work only with the hash type data (`{}`) not with strings
2. If the file data is a hash with one root key: it will replace the root key of the source data with one from the restored data.
- That's also okay, since the source data and the redacted data should have the same key structure

Reply:
It is true that the first key of a JSON file would still be restored, although I feel it is not something we want to do by design.
I'm also thinking long-term, when/if we move content from YAML files to JSON files, this exception won't be needed anymore, in fact, this are more reasons to stop using YAML.
Not a blocker, but if we had a condition that checked the file extension, it would be easier to remove the YAML exception down the road.